README.adoc

```diff
@@ -80,25 +80,52 @@ checkout([$class: 'GitSCM',
 
 ==== Checkout with ssh and a private key credential
 
-Insert example here
+Checkout from the git client plugin source repository using https protocol, ssh private credentials, and a non-default branch. 
+The git plugin supports private key credentials provided by the link:https://plugins.jenkins.io/credentials[Jenkins credentials plugin].
+
+[source,groovy]
+---
+checkout([$class: 'GitSCM',
+          branches: [[name: 'master']],
+          userRemoteConfigs: [[credentialsId: 'e1476d45-b77d-4af6-a80f-e2fe95eca822', url: 'https://github.com/jenkinsci/git-plugin.git']]])
+---
 
 ==== Checkout with https and a username/password credential
 
-Insert example here
+Checkout from the git client plugin source repository using https protocol, username/password credentials, and a non-default branch. The git plugin supports username/password credentials provided by the link:https://plugins.jenkins.io/credentials[Jenkins credentials plugin].
+
+[source,groovy]
+---
+checkout([$class: 'GitSCM',
+          branches: [[name: 'jenkinsfile']],
+          userRemoteConfigs: [[credentialsId: 'e1476d45-b77d-4af6-a80f-e2fe95eca822', url: 'https://github.com/jenkinsci/git-client-plugin.git']]])
```

Comment:
Best to use a branch that exists in the repository.
```suggestion
          branches: [[name: 'stable-2.x']],
          userRemoteConfigs: [[credentialsId: 'my-username-password',
                             url: 'https://github.com/jenkinsci/git-client-plugin.git']]])
```

Reply:
okay. Will do that.